### ISSUE TITLE
Update pagination, get AJAX working again

### ISSUE THREAD

**Issue by mjy** [mx3/mx#8]
We didn't jump on the will_paginate bandwagon, but perhaps we should now. 

Our existing AJAX pagination is fairly straightforward - we create our own set of anchors (helpers/app/navigation_helper/pagination_links_with_ajax(paginator, options={}) using classic pagination.  The page partial is _ajax_list, it's called/replaced in list.html.erb.  This pattern is repeated in nearly all controllers.  

Fix the functionality for OTUs first, then we'll likely throw a "grind" label on this for our side to spawn across the other controllers.

Render the pagination links with a similar helper method that will be generic for all models.

Basic Requirements:
- maintain the same functionality and layout (next/previous must be beside each other) as exists in edge/production
- When AJAX not available pagination should fall back gracefully to just rendering whole pages

Bonus/other Considerations (if relatively trivial additions)
- experiment with infinite scrolling as possible strategy for replacement of pages
- experiment with sortable columns (within page only)
- provide "jump to page" popup form functionality for navigation cluster
- make the pagination code modular enough to ultimate layer filters over top the pagination result (this will likely become a new ticket once this is finished.

**Comment by caryfitzhugh** [comment]
Could you give some more detail on what you mean by 'jump to page' popup form functionality?

And some more detail on what it would look like to layer filters over top of the pagination? 
Does that mean to allow there to be search filters (kindof like www.ziplist.com/recipes/search in the left column?)

**Comment by mjy** [comment]
Re filters- exactly like that.  

Re "jump to page".  I can't find the example I as thinking of- but if you imagine a tool-tip like popup when you hover over next/previous, and that popup has a form with one box, you can type in a page number, and click the go button beside it to navigate the pagination directly to that page.  Don't make it a priority (a faceted browse/filter would be much handier).

Let's make filters/filters over pagination a new issue. Seems to warrant enough discussion on it's own.  Assume this is the basic pagination (as implemented in current production) issue.

**Comment by caryfitzhugh** [comment]
Facets are a big deal.  That would probably take up a good bit of time to
implement.
I have some ideas on how I'd like to do it this time (based on how painful
it turned out to be on the Ziplist site)...
but that's probably a longer-term improvement.. but one that usually is a
really good feature.

The approach I'd take would basically be to build a simplistic
inverted/reverse index in redis, and query that, instead of querying a text
indexer (we used sphinx). Redis has a lua extension, so we could do all the
heavy lifting inside of redis.  It'd be blazing fast.

It does depend on how many records you are filtering through - and if you
want facet counts or not.  If you don't want counts, there may be cheaper
options.

Anyway -- tis for another day. :)

**Comment by mjy** [comment]
Go ahead and get the basic AJAX pagination working again, as it presently exists, without facets/filters, and without the jump to page" feature.

**Comment by mjy** [comment]
I think we should run with kaminari.  Looks like the new hotness.

**Comment by caryfitzhugh** [comment]
Kaminari is in - classic_pagination is out.

Most controllers will probably fail now though - but Otus works ok.

I'm going to walk through a few more controllers shortly and 'bring them up to speed'.

**Comment by caryfitzhugh** [comment]
closed.
